### PR TITLE
refactor plugin to add event hierarchy for different event types

### PR DESCRIPTION
**Pull request by guotuofeng**
Add event classes hierarchy for different event types like: kernel, operator, duration, etc.

### CODE REVIEW

File: tb_plugin/torch_tb_profiler/profiler/trace.py [hunk 1]
```diff
@@ -45,36 +45,41 @@ def __init__(self, type, data):
...
+        self.input_shape = shape
 
-        return shape
-
-    @property
-    def input_type(self):
-        return self.args.get("Input type")
-
-class ProfilerStepEvent(TraceEvent):
+class ProfilerStepEvent(OperatorEvent):
```
Comment (gaoteng-git) [inline tb_plugin/torch_tb_profiler/profiler/trace.py:82]: I'm not sure why its inherits changed from DurationEvent to OperatorEvent.

**Review by gaoteng-git — COMMENTED** [review]
As events could be distinguished by class type, such as "OperatorEvent", "KernelEvent", is it possible to remove BaseEvent.type?

**Review by gaoteng-git — APPROVED** [review]
LGTM.

File: tb_plugin/torch_tb_profiler/profiler/trace.py [hunk 2]
```diff
@@ -124,8 +122,14 @@ def create_trace_event(event):
         name = event.get("name")
         if name and name.startswith("ProfilerStep#"):
             return ProfilerStepEvent(event)
-
-    if event_type is not None:
-        return TraceEvent(event_type, event)
+        else:
+            return OperatorEvent(event_type, event)
+    elif event_type == EventTypes.PYTHON:
```
Comment (guyang3532) [inline tb_plugin/torch_tb_profiler/profiler/trace.py:127]: If change EventTypeMap['Python'] = EventTypes.OPERATOR, then this elif can be removed.

**Pull request merged**